packages/mui-base/src/utils/useControlled.test.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { act, createRenderer } from '@mui/internal-test-utils';
+import { useControlled } from './useControlled';
```

Comment:
What if we were to already expose them as a new npm package https://www.notion.so/mui-org/engineering-mui-utils-purpose-9a9fc9da3a004864b6c4e1f4d1f24f95?p=56e98feced4d4fe4a73df99e0b0361b4&pm=s while we are at it? so that Material UI, MUI X, can migrate to those and @mui/utils can deprecate those utils?

Or maybe it's too big of a jump at once?

Reply:
Let's do it incrementally. As I noted in https://www.notion.so/engineering-mui-utils-purpose-9a9fc9da3a004864b6c4e1f4d1f24f95?d=143cbfe7b66080e1ad16001c70691469&pvs=4#7cb73e22aaee40eb9c35cca853ffc73f, I'm not particularly keen on placing this kind of utilities in a separate package. I think they are useful to build styled libraries and should be exposed from @base-ui-components/react directly.